dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -522,7 +538,18 @@ public Response submitMetricsData(String id, String versionId, Partner platform,
     }
 
     @Override
-    public Response setAggregatedMetrics(String id, String versionId, Partner platform, Metrics aggregatedMetrics) {
+    public Response getEntryVersionsToAggregate() {
+        List<EntryLiteAndVersionName> entryAndVersionNames = new ArrayList<>();
+        entryAndVersionNames.addAll(toolDAO.findEntryVersionsToAggregate());
+        entryAndVersionNames.addAll(bioWorkflowDAO.findEntryVersionsToAggregate());
+        entryAndVersionNames.addAll(notebookDAO.findEntryVersionsToAggregate());
+        entryAndVersionNames.addAll(appToolDAO.findEntryVersionsToAggregate());
+        entryAndVersionNames.addAll(serviceDAO.findEntryVersionsToAggregate());
+        return Response.ok(entryAndVersionNames).build();
+    }
+
+    @Override
+    public Response setAggregatedMetrics(String id, String versionId, Map<Partner, Metrics> aggregatedMetrics) {
```

Comment:
The PUT aggregated metrics endpoint that we use internally now takes a Map of metrics instead of a platform and `Metrics` object. This endpoint now overwrites the entire Version metrics map with the one provided in request body. The reason is so the `latestMetricsAggregationDate` is set only updated, when all platform aggregated metrics are submitted in one request call instead of each time a platform aggregated metric is submitted.

Recall: the metrics aggregator previously submitted

Reply:
Not sure I follow, could you re-phrase?

Reply:
Ha, just noticed the unfinished "Recall" sentence :sweat_smile: 

For a version, the metrics aggregator will aggregate metrics for every platform that the version has metrics for and submit each platform metric one at a time. Suppose that a version has metrics for Terra and DNAstack. The metrics aggregator will aggregate metrics for Terra, post it to the webservice, aggregate metrics for DNAstack, post it to the webservice, aggregate metrics across all platforms then post it to the webservice. That's three POSTs for one version. 

The difficulty with this is differentiating when all of the platform metrics have been submitted for a version so that we can update the `latestMetricsAggregationDate`. If we set it too soon and the metrics aggregator dies mid-way through aggregating the platform metrics for a version, then subsequent calls to the new `getEntryVersionsToAggregate` will not return this version and the version will be in a "half" aggregated state where it has metrics for some platforms but potentially not all.

That's the reason I updated this endpoint so it takes all platform metrics at once.

Reply:
Ok, I see. The trade-off is that we would need to process all platforms at once and upload them (which seems ok), removing the ability to update one platform at a time. Otherwise, we'd need to track dates for each combination of platforms and a particular version which indeed sounds painful. 

Makes sense

Reply:
> The trade-off is that we would need to process all platforms at once and upload them (which seems ok)

With Athena, we already do this because the query has a `group by platform` statement, so should be no real difference from the metrics-aggregator side